usaspending_api/idvs/v2/views/amounts.py

```diff
@@ -9,12 +10,98 @@
 from usaspending_api.common.validator.tinyshield import TinyShield
-
+from usaspending_api.awards.v2.data_layer.sql import defc_sql
 
 logger = logging.getLogger("console")
 
+child_award_sql = """
+    select
+        ac.id as award_id
+    from
+        parent_award pap
+        inner join awards ap on ap.id = pap.award_id
+        inner join awards ac on ac.fpds_parent_agency_id = ap.fpds_agency_id and ac.parent_award_piid = ap.piid and
+            ac.type not like 'IDV%'
+    where
+        pap.{award_id_column} = '{award_id}'
+    """
+grandchild_award_sql = """
+    select
+        ac.id as award_id
+    from
+        parent_award pap
+        inner join parent_award pac on pac.parent_award_id = pap.award_id
+        inner join awards ap on ap.id = pac.award_id
+        inner join awards ac on ac.fpds_parent_agency_id = ap.fpds_agency_id and ac.parent_award_piid = ap.piid and
+            ac.type not like 'IDV%'
+
+    where
+        pap.{award_id_column} = '{award_id}'
+    """
+
+
+def fetch_account_details_idv(award_id, award_id_column) -> dict:
+    if award_id_column != "award_id":
+        award_id = re.sub(r"[']", r"''", award_id)
+    children = execute_sql_to_ordered_dictionary(
+        child_award_sql.format(award_id=award_id, award_id_column=award_id_column)
+    )
+    grandchildren = execute_sql_to_ordered_dictionary(
+        grandchild_award_sql.format(award_id=award_id, award_id_column=award_id_column)
+    )
+
+    child_award_ids = []
+    grandchild_award_ids = []
+    child_award_ids.extend([x["award_id"] for x in children])
+    grandchild_award_ids.extend([x["award_id"] for x in grandchildren])
+    award_id_sql = "faba.award_id in {award_id}".format(award_id=child_award_ids)
+    child_results = (
+        execute_sql_to_ordered_dictionary(defc_sql.format(award_id_sql=award_id_sql)) if child_award_ids != [] else {}
+    )
+    award_id_sql = "faba.award_id in {award_id}".format(award_id=grandchild_award_ids)
+    grandchild_results = (
+        execute_sql_to_ordered_dictionary(defc_sql.format(award_id_sql=award_id_sql))
+        if grandchild_award_ids != []
+        else {}
+    )
+    child_outlay_by_code = []
+    child_obligation_by_code = []
+    child_total_outlay = 0
+    child_total_obligations = 0
+    for row in child_results:
+        child_total_outlay += row["total_outlay"]
+        child_total_obligations += row["obligated_amount"]
+        child_outlay_by_code.append({"code": row["disaster_emergency_fund_code"], "amount": row["total_outlay"]})
+        child_obligation_by_code.append(
+            {"code": row["disaster_emergency_fund_code"], "amount": row["obligated_amount"]}
+        )
+    grandchild_outlay_by_code = []
+    grandchild_obligation_by_code = []
+    grandchild_total_outlay = 0
+    grandchild_total_obligations = 0
+    for row in grandchild_results:
+        grandchild_total_outlay += row["total_outlay"]
+        grandchild_total_obligations += row["obligated_amount"]
```

Comment:
Same problem as above. Totals need to be for covid-only DEFCs

Reply:
I was under the impression that the totals where just the total spending we had for all DEFC, and the frontend would be filtering out which DEFC they wanted to use on their end.
But actually thinking about it it doesn't make much sense to do it that way, because how would they know which DEFC are covid-related, so I will in fact change this

Reply:
The website is using the reference endpoint which does give the necessary info on which DEFC are COVID-19
https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/api_contracts/contracts/v2/references/def_codes.md